Sources_v3/Workers/Background/MessageEditor.swift

```diff
@@ -0,0 +1,108 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import CoreData
+import Foundation
+
+class MessageEditor<ExtraData: ExtraDataTypes>: Worker {
+    @Atomic private var pendingMessageIDs: Set<MessageId> = []
+    private let observer: ListDatabaseObserver<MessageDTO, MessageDTO>
+    private let sendingQueue: DispatchQueue
+
+    override init(database: DatabaseContainer, webSocketClient: WebSocketClient, apiClient: APIClient) {
+        observer = .init(
+            context: database.backgroundReadOnlyContext,
+            fetchRequest: MessageDTO.messagesPendingEditFetchRequest(),
+            itemCreator: { $0 }
+        )
+        
+        sendingQueue = .init(
+            label: "co.getStream.ChatClient.MessageEditorQueue",
+            qos: .userInitiated
+        )
+        
+        super.init(database: database, webSocketClient: webSocketClient, apiClient: apiClient)
+        
+        startObserving()
+    }
+}
+
+// MARK: - Private
+
+private extension MessageEditor {
```

Comment:
Why have this in an extension?

Reply:
When we declare methods in `private extensions` we don't need to mark each one as `private`